### PR TITLE
add stylelint

### PR DESCRIPTION
**Pull request by RyanHirsch**
This adds stylelint as a CI only check and corrects some of the issues it flagged. I also ran prettier across the files to clean up a bunch of whitespace issues, github provides an option to hide these when reviewing the code
Screen Shot 2022-01-07 at 22 20 48

### CODE REVIEW

File: styles/mixins/themes.scss [hunk 1]
```diff
@@ -109,7 +109,7 @@ $scopedTheme: null;
             $scopedTheme: () !global;
             @each $variableName, $variableValue in $themeMap {
                 // Merge each key-value pair into the theme specific map
-                $scopedTheme: map-merge($scopedTheme, ($variableName: $variableValue)) !global;
+                $scopedTheme: map.merge($scopedTheme, ($variableName: $variableValue)) !global;
```
Comment (RyanHirsch) [inline styles/mixins/themes.scss:112]: It looks like both of these are technically correct according to https://sass-lang.com/documentation/modules/map#merge

The linter seems to prefer the `map.merge` version as do the examples on https://sass-lang.com/documentation/values/maps#add-to-a-map

File: styles/components/index.scss [hunk 2]
```diff
@@ -1,79 +1,78 @@
...
+@import "../../src/components/FundingLink/FundingLink";
+@import '../../src/components/EpisodePageHeader/EpisodePageHeader';
+@import '../../src/components/Icon/Icon';
+@import '../../src/components/MainContentSection/MainContentSection';
+@import '../../src/components/MakeClip/MakeClipForm';
+@import '../../src/components/MakeClip/MakeClipModal';
+@import '../../src/components/MakeClip/MakeClipSuccessModal';
+@import '../../src/components/MediaItemControls/MediaItemControls';
+@import '../../src/components/HorizontalNavBar/HorizontalNavBar';
+@import '../../src/components/NavBar/Mobile/MobileNavBar';
```
Comment (RyanHirsch) [inline styles/components/index.scss:24]: This was actually double imported in the existing file, the linter found it.

File: .stylelintrc.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "stylelint-config-recommended-scss",
+  "rules": {
+    "block-no-empty": null,
+    "no-empty-source": null,
+    "no-descending-specificity": null
```
Comment (RyanHirsch) [inline .stylelintrc.json:6]: I don't have opinions on these, but the linter didn't like them, so I just disabled to keep the current standards in place.

File: styles/mixins/themes.scss (hunk 1 above)
Comment (RyanHirsch) [reply styles/mixins/themes.scss]: Oh and there's also this - https://css-tricks.com/introducing-sass-modules/

**Review by mitchdowney — APPROVED** [review]

**Pull request merged**